nir/src/main/scala/scala/scalanative/nir/Arg.scala

```diff
@@ -0,0 +1,40 @@
+package scala.scalanative.nir
+
+/**
+ * Created by Kamil on 23.07.2016.
+ */
+case class Arg(ty: Type, attrs: ArgAttrs = ArgAttrs.empty)
+
+case class ArgAttrs(byval: Option[Type] = None, sret: Option[Type] = None) {
```

Comment:
Lets just inline those fields into `Arg` itself. Having an alternative seq-based representation seems to be an overkill here. Similarly, current design for `nir.Attrs` is quite an overkill and we'll probably simplify it in the future, but that's a problem for another PR.